easybuild/easyblocks/f/flang.py

```diff
@@ -0,0 +1,186 @@
+            raise EasyBuildError(
+                "Could not determine LLVM source root (LLVM source was not unpacked?)")
+        src_dirs = {}
+
+        def find_source_dir(globpatterns, targetdir):
+            """Search for directory with globpattern and rename it to targetdir"""
+            if not isinstance(globpatterns, list):
+                globpatterns = [globpatterns]
+
+            glob_src_dirs = [glob_dir for globpattern in globpatterns for glob_dir in
+                             glob.glob(globpattern)]
+            if len(glob_src_dirs) != 1:
+                raise EasyBuildError(
+                    "Failed to find exactly one source directory for pattern %s: %s",
+                    globpatterns,
+                    glob_src_dirs)
+            src_dirs[glob_src_dirs[0]] = targetdir
+
+        if LooseVersion(self.version) >= LooseVersion('3.8'):
+            find_source_dir('openmp-*',
+                            os.path.join(self.llvm_src_dir, 'projects', 'openmp'))
+
+        find_source_dir('flang-driver-*',
+                        os.path.join(self.llvm_src_dir, 'tools', 'clang'))
+
+        # Place the flang code in a separate directory for the build step for after
+        # we've built llvm
+        self.flang_source_dir = os.path.join(self.llvm_src_dir, '..', 'flang')
+        find_source_dir('flang-flang_*', self.flang_source_dir)
+
+        for src in self.src:
+            for (dirname, new_path) in src_dirs.items():
+                if src['name'].startswith(dirname):
+                    old_path = os.path.join(src['finalpath'], dirname)
+                    try:
+                        shutil.move(old_path, new_path)
+                    except IOError as err:
+                        raise EasyBuildError("Failed to move %s to %s: %s", old_path,
+                                             new_path, err)
+                    src['finalpath'] = new_path
+                    break
+
+    def build_with_temporary_llvm(self, build_dir, src_dir, parallel=True, additional_options=[]):
+        """Build Clang stage N using Clang stage N-1"""
+
+        # Create and enter build directory.
+        mkdir(build_dir)
+        change_dir(build_dir)
+
+        # Configure.
+        CC = os.path.join(self.llvm_obj_dir, 'bin', 'clang')
+        CXX = os.path.join(self.llvm_obj_dir, 'bin', 'clang++')
+        FC = os.path.join(self.llvm_obj_dir, 'bin', 'flang')
+        LLVM_CONFIG = os.path.join(self.llvm_obj_dir, 'bin', 'llvm-config')
+        shlib_ext = get_shared_lib_ext()
+        LIBOMP = os.path.join(self.llvm_obj_dir, 'lib', 'libomp.%s' % shlib_ext)
+
+        options = "-DCMAKE_INSTALL_PREFIX=%s " % self.installdir
+        options += "-DCMAKE_C_COMPILER='%s' " % CC
+        options += "-DCMAKE_CXX_COMPILER='%s' " % CXX
+        options += "-DCMAKE_Fortran_COMPILER='%s' " % FC
+        options += "-DLLVM_CONFIG='%s' " % LLVM_CONFIG
+        options += "-DFLANG_LIBOMP='%s' " % LIBOMP
```

Comment:
If we added Clang as a dependency I think it would be possible to just point to that `libomp` at this point (and not build Flang `libomp` at all).

Reply:
We would likely want Clang+Flang as a single entity, just like we just merged icc/ifort
It makes for a much improved user experience not to have to understand that both icc and ifort have to be loaded despite Lmods spider saying you only need one...

Reply:
For Flang, I don't think that is actually a good idea. Flang is (unfortunately) working off a forked version of Clang.

This is likely to very soon be out of date anyway...though I admit to being confused by the whole thing: `f18` was supposed to replace `flang` but it seems it will actually take over the name `flang` (see [here](https://www.phoronix.com/scan.php?page=news_item&px=LLVM-f18-Will-Be-Flang)). Once that get's a release then it is very likely that incorporating flang will become a whole lot easier...it is probably best to wait it out.